packages/astro/src/integrations/index.ts

```diff
@@ -97,6 +99,19 @@ export async function runHookConfigSetup({
 				addWatchFile: (path) => {
 					updatedSettings.watchFiles.push(path instanceof URL ? fileURLToPath(path) : path);
 				},
+				addClientDirective: ({ name, entrypoint }) => {
+					if (!settings.config.experimental.customClientDirectives) {
+						throw new Error(
```

Comment:
Can/should we be using `AstroError` here? cc @Princesseuh

Reply:
Nope, since this particular error will be removed when it'll be unflagged. Temporary errors and errors in integrations are the two that don't / shouldn't use AstroError